lib/swaps/Swaps.ts

```diff
@@ -141,207 +132,218 @@ class Swaps extends EventEmitter {
       makerCurrency = baseCurrency;
     }
     const { takerAmount, makerAmount } = Swaps.calculateSwapAmounts(taker.quantity, maker.price);
+    const preimage = randomBytes(32);
+    const hash = createHash('sha256');
+    const r_hash = hash.update(preimage).digest('hex');
 
-    const dealRequestBody: packets.DealRequestPacketBody = {
-      dealId,
+    const swapRequestBody: packets.SwapRequestPacketBody = {
       takerCurrency,
       makerCurrency,
+      r_hash,
       takerAmount,
       makerAmount,
-      takerPubKey: takerCurrency === 'BTC' ? this.pool.handshakeData.lndbtcPubKey! : this.pool.handshakeData.lndltcPubKey!,
       orderId: maker.id,
       pairId: maker.pairId,
       proposedQuantity: taker.quantity,
     };
 
     const deal: SwapDeal = {
-      ...dealRequestBody,
-      myRole : SwapDealRole.Taker,
-      peerPubKey: this.pool.handshakeData.nodePubKey,
+      ...swapRequestBody,
+      r_preimage: preimage.toString('hex'),
+      myRole: SwapDealRole.Taker,
       createTime: Date.now(),
     };
 
     this.addDeal(deal);
-    this.logger.debug('Initiating swap deal: ' + JSON.stringify(deal));
+    this.logger.debug('[SWAP] Initiating deal: ' + JSON.stringify(deal));
```

Comment:
No need for the `[SWAP]` here as this is already inserted into the logs from the swaps context.